### ISSUE TITLE
Increase hitSlop and pressRetentionOffset for navbar buttons

### ISSUE THREAD

**Issue by ghost** [react-navigation/stack#36]
This is an issue from react-navigation https://github.com/react-navigation/react-navigation/issues/287
As ide said in that issue, we could just increase the hitSlop in the header back button to make it easier to touch. I just looked at the current code and currently on iOS we're using margin to increase the touch area instead of using hitSlop.
https://github.com/react-navigation/react-navigation-stack/blob/106f1e74db6fa66b27f42d12648fe22d8741fd1b/src/views/Header/HeaderBackButton.js#L156
So that's another way to make the button easier to touch and I think it would make the implementation between android and iOS more consistent but that would also require a little more code.
What do you guys think?

**Comment by JensDebergh** [comment]
@tienf4m @brentvatne Did you guys find any solution to increasing the touch area with current version of `react-navigation`?

We are currently having a similar issue where we only changed the back button image. Everything works properly on iOS, but on android the hit area is to small.

**Comment by brentvatne** [comment]
@JensDebergh - we'll be able to improve this on android on the next release of react-native-gesture-handler - with it we will change to using react-native-gesture-handler for the button (we have on ios already) but we had to old off temporarily because it didn't yet support custom ripple colors on android. then we can increase hitslop: https://github.com/react-navigation/react-navigation-stack/blob/master/src/views/TouchableItem.js#L57

**Comment by JensDebergh** [comment]
@brentvatne Thanks for the heads up!

For anyone wondering how to get the same effect on android without completely overriding the `headerLeft` property of `react-navigation`.

I played around with margins on android to get the desired effect like so:

1) Add margins to `headerBackImage` to increase overall touch area of the back button on all platforms.

2) Add negative `marginLeft` for the same amount as the added margins on the `backImage` property.

3) Extract navigation options in a separate object for easy access throughout the app.

You'll have to fiddle around with those 2 values to get the desired result in your app. But we got it working on both android and ios with the same touch surface.


```
	headerLeftContainerStyle: {
		marginLeft: Utils.isAndroid() ? -8 : 0, <-- Pull image to the left for 8 pixels on android
	},
	headerBackImage: (
		<Image
			source={Images.icArrowBack16()}
			style={{
				width: 16,
				height: 16,
				tintColor: Style.color.trout,
				margin: Utils.isAndroid() ? 8 : 16, <-- Add margin to the image to increase touch area
			}}
		/>
	),
```
